### ISSUE TITLE
Cannot delete the first workspace while deletion icon is active

### ISSUE THREAD

**Issue by bramhoven** [usebruno/bruno#38]
I don't know whether this is a bug or intended behaviour, but either way something goes wrong.

The first workspace cannot be deleted like the other workspaces while it can be renamed.
![](https://i.imgur.com/QWYAwPZ.gif)
If it cannot be deleted, the trash icon should not be shown in my opinion.

**Comment by bramhoven** [comment]
Ah ofcourse! Did not even notice that haha. I could also submit a PR to update the error message if you are busy with other stuff
Really like bruno so far, keep the progress going!

**Comment by helloanoop** [comment]
@bramhoven That'd be awesome.

Here is the [guide](https://github.com/usebruno/bruno/blob/main/contributing.md) to get started and bring bruno on your local

To save you some time, [here](https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/components/Workspaces/WorkspaceConfigurer/DeleteWorkspace/index.js#L17) is where the generic error message is being thrown and [here](https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/providers/ReduxStore/slices/workspaces/actions.js#L113) the error gets thrown

I have some code already baked in to implement generic error handling. Its [here](https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/utils/common/error.js#L20)

The way I would approach this it throw a `BrunoError` in workspaces actions, and then have the `DeleteWorkspace` use the `toastError` 

I also, think it'd be good to update `toastError` to support a default error
```js
export const toastError = (error, defaultErrorMsg = 'An error occurred') => {
  if (error instanceof BrunoError) {
    if (error.level === 'warning') {
      return toast(error.message, {
        icon: '⚠️',
        duration: 3000
      });
    }
    return toast.error(error.message, {
      duration: 3000
    });
  }

  return toast.error(error.message || defaultError);
```

**Comment by bramhoven** [comment]
Trying to run bruno, but getting this error.
```sh
bruno on  feature/improve-error-workspace-deletion via  v14.17.0 
❯ npm run dev:web

> usebruno@ dev:web C:\workspaces\bruno
> npm run dev --workspace=packages/bruno-app

npm ERR! missing script: dev

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\bram.hoven\AppData\Roaming\npm-cache\_logs\2022-10-21T15_01_03_235Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! usebruno@ dev:web: `npm run dev --workspace=packages/bruno-app`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the usebruno@ dev:web script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\bram.hoven\AppData\Roaming\npm-cache\_logs\2022-10-21T15_01_03_278Z-debug.log
```

Currently the scripts in `package.json` are:
```json
"scripts": {
  "dev:web": "npm run dev --workspace=packages/bruno-app",
  "build:web": "npm run build --workspace=packages/bruno-app",
  "dev:electron": "npm run dev --workspace=packages/bruno-electron",
  "build:chrome-extension": "./scripts/build-chrome-extension.sh",
  "build:electron": "./scripts/build-electron.sh"
}
```

So it is missing the dev script required to run `npm run dev --workspace=packages/bruno-app`

**Comment by helloanoop** [comment]
Hmm.. strange..
Can you check if the npm version is 8.x
This error is common in older npm versions that do not support workspaces

If the issue still persists, then for now, you can
```bash
cd packages/bruno-app
npm run dev
```

**Comment by bramhoven** [comment]
I was running version 6.14. I am upgrading now. Thanks for the quick response!

**Comment by bramhoven** [comment]
@helloanoop I created PR #39. If you have the time let me know what you think about it!
I also added that the error messages is gotten through the `parseError` method since that seemed specifically made for getting the correct message from an error.

**Comment by helloanoop** [comment]
Merged !
Thanks for this @bramhoven !

The fix should be deployed in the web version https://play.usebruno.com/ in the next couple of min.
Will be cutting a desktop release with these fixes tomorrow.

**Comment by bramhoven** [comment]
Great! I did see the Vercel build had failed... I was not able to see why it had failed though

**Comment by helloanoop** [comment]
Yup. Some deployment config got updated in the vercel settings.
This is fixed now. The latest build is now deployed to https://play.usebruno.com

**Comment by bramhoven** [comment]
Cool! Tested it and it seems to work 😁

**Comment by helloanoop** [comment]
@bramhoven Release v0.2.0 with your fixes is out. Thanks again for the work on this.
You can download the latest versions at https://www.usebruno.com/downloads

In case you love dark more, its included in the v0.2.0 release
Currently its based on vscode dark theme, but plan to support custom themes in the near future.
